packages/trpc/server/createNextApiHandler.ts

```diff
@@ -58,19 +58,36 @@ export function createNextApiHandler(router: AnyRouter, isPublic = false, namesp
 
       if (isPublic && paths) {
         const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+        const FIVE_MINUTES_IN_SECONDS = 5 * 60;
         const cacheRules = {
-          session: `no-cache`,
-          i18n: `no-cache`,
-          // Revalidation time here should be 1 second, per https://github.com/calcom/cal.com/pull/6823#issuecomment-1423215321
-          "slots.getSchedule": `no-cache`, // FIXME
-          cityTimezones: `max-age=${ONE_DAY_IN_SECONDS}, stale-while-revalidate`,
-          "features.map": `max-age=${ONE_DAY_IN_SECONDS}, stale-while-revalidate`, // "map" - Feature Flag Map
+          session: "no-cache",
+
+          // i18n data is user specific and thus should not be cached
+          i18n: "no-cache",
+
+          // We want getSchedule to be fresh always, but we would let it be stale for 60 seconds in worst case without impacting the user experience
+          // Note that, after 1 second, if getSchedule takes say 0.1s to respond, a request sent after 1.1s would get the fresh value of getSchedule.
+          "slots.getSchedule": `max-age=1, stale-while-revalidate=60`,
+
+          // Timezones are hardly updated. No need to burden the servers with requests for this by keeping low max-age.
+          // Keep it cached for a day and then give it 60 seconds more at most to be updated.
+          cityTimezones: `max-age=${ONE_DAY_IN_SECONDS}, stale-while-revalidate=60`,
+
+          // Feature Flags change but it might be okay to have a 5 minute cache to avoid burdening the servers with requests for this.
+          // Note that feature flags can be used to quickly kill a feature if it's not working as expected. So, we have to keep fresh time lesser than the deployment time atleast
```

Comment:
```suggestion
          // Note that feature flags can be used to quickly kill a feature if it's not working as expected. So, we have to keep stale time lesser than the deployment time atleast, so that after 5 minutes the changes are updated.
```